versioned/tiered/src/main/java/com/dremio/nessie/tiered/builder/L1Builder.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.tiered.builder;
+
+import com.dremio.nessie.versioned.Key;
+import com.dremio.nessie.versioned.store.Id;
+
+/**
+ * Interface to create an L1 Builder. To be implemented by each {@link Store} implementation.
```

Comment:
I like the idea to have such a thing, as that effectively ensures that each store implements all aspects of an L1 (compared to something like `Store.storeL1(L1 l1)` and having the store iterate through the L1).
The javadoc should mention the "dual use" aspect as well, since it's used in `L1.builder()` as well, so for/from "both sides".